### PR TITLE
chore: Ts strict remaining

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://github.com/adobe/react-spectrum/issues/3183

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

### CODE REVIEW

File: packages/@react-spectrum/tree/src/TreeView.tsx [hunk 1]
```diff
@@ -333,6 +333,7 @@ const expandButton = style<ExpandableRowChevronProps>({
 
 function ExpandableRowChevron(props: ExpandableRowChevronProps) {
   let expandButtonRef = useRef<HTMLSpanElement>(null);
+  // @ts-ignore - not sure how to type this or convert to a button (it fails a test if i convert)
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-spectrum/tree/src/TreeView.tsx:336]: looks like we can just remove the type for the expandButtonRef... either that or ignore or cast it to any.

File: packages/@react-aria/menu/src/useMenuTrigger.ts [hunk 2]
```diff
@@ -128,6 +128,7 @@ export function useMenuTrigger<T>(props: AriaMenuTriggerProps, state: MenuTrigge
   delete triggerProps.onPress;
 
   return {
+    // @ts-ignore - TODO we pass out both DOMAttributes AND AriaButtonProps, but useButton will discard the longPress event handlers, it's only through PressResponder magic that this works for RSP and RAC. it does not work in aria examples
```
[the remark this answers is not in the document]
Comment (devongovett) [reply packages/@react-aria/menu/src/useMenuTrigger.ts:131]: I think we'll have to leave the ts-ignore for now...

**Review by reidbarber — DISMISSED** [review]
Amazing!

**Review by LFDanLu — DISMISSED** [review]
Did a sweep of virtualized components, overlay positioning, and the generated TS diff and things seemed to be working as expected. Approving for testing

**Review by LFDanLu — APPROVED** [review]

**Review by reidbarber — APPROVED** [review]

**Pull request merged**